mozdownload/scraper.py

```diff
@@ -36,7 +36,10 @@
 APPLICATIONS = ('b2g', 'firefox', 'thunderbird')
 
 # Base URL for the path to all builds
-BASE_URL = 'https://ftp.mozilla.org/pub/mozilla.org'
+BASE_URL = 'http://download.cdn.mozilla.net/pub'
+
+# In case CDN fails
+BACKUP_URL = 'https://ftp.mozilla.org/pub/mozilla.org'
```

Comment:
You weren't actually forwarded to ftp.m.o in the sense of a 302. The headers you get from a request to the CDN are sum of
* the request from CDN to origin server
* your request to the CDN
The origin server is a vhost on the same cluster that serves ftp.m.o, so you get this sort of thing
  X-Backend-Server: ftp5.dmz.scl3.mozilla.com
when making a CDN request, and it doesn't mean you talked to ftp.m.o directly.

The origin server for https://download-installer.cdn.mozilla.net blocks downloads of .mar files at the time of download, but doesn't modify directory listings. eg if you try to download
 https://download-installer.cdn.mozilla.net/pub/firefox/releases/36.0/update/win32/en-US/firefox-34.0.5-36.0.partial.mar
you get a 403.